avalanche/caip2.md

```diff
@@ -0,0 +1,126 @@
+---
+namespace-identifier: avax-caip2
+title: Avalanche Namespace - Chains
+author: Gergely Lovas (@gergelylovas)
+discussions-to: []
+status: Draft
+type: Standard
+created: 2024-07-16
+requires: CAIP-2
+---
+
+# CAIP-2
+
+*For context, see the [CAIP-2][] specification.*
+
+## Rationale
+
+In CAIP-2 a general blockchain identification scheme is defined. This is the
+implementation of CAIP-2 for the Avalanche Ecosystem. 
+Blockchains in the "avax" namespace are validated by their `blockchainID`. The `blockchainID` in Avalanche refers to the [txID][] that created the blockchain on the Avalanche P-Chain.
+These blockchain IDs require transformations to be used as conformant CAIP-2
+references.
+
+## Syntax
+
+The namespace "avax" refers to the Avalanche open-source blockchain platform.
+
+### Reference Definition
+
+The definition for this namespace will use the `blockchainID` as an identifier
+for different Avalanche chains. Since the P-Chain's `blockchainID` is the same on Testnet and Mainnet, to ensure uniqueness, a testnet prefix is introduced for chains on the [Fuji Testnet][].  
+
+The method for calculating the chain ID is as follows with pseudo-code:
+
+```
+hex(sha256(concat(testnet_prefix, blockchain_id)))
+```
+
+- `blockchain_id` = `txID` that created the blockchain on the Avalanche P-Chain
+- `testnet_prefix`= `fuji` when the chain was created on the Fuji Testnet; empty string otherwise
+- `concat`= a string concatenation function
+- `sha256`= a SHA256 hash function
+- `hex`= a lowercase hex encoder
+- `first_32_chars`= a function to extract the first 32 characters of the
```

Comment:
`first_32_chars` is not used anywhere in the above formula

Reply:
🤦 fixed